### PR TITLE
[docs] add "Components" section to the generated docs, migrate `AppleAuthenticationButton`

### PR DESCRIPTION
**Pull request by Simek**
# Why & How

Refs ENG-910

This PR adds the missing "Components" section to the API auto generation feature.

During the development of this section I have use the `AppleAuthenticationButton` component from `expo-apple-authentication` package, which docs are now fully auto generated.

I have also applied small simplifications in the `AppleAuthenticationButton` code, which should not require the changelog entry (but I'm not 100% sure about `FunctionComponent` => `FC` type change, it should be threaten as an alias, but please let me know if this change should be listed).

I'm also not sure if the `AppleAuthenticationButtonProp` type should be extended by `React.ComponentProps<typeof View>`, as the docs suggest in of of paragraphs:
> The properties of this component extend from `View`...

The changes have been only applied to the `unversioned` docs, please let me know if I should backport this to the SDK 42 docs.

# Test Plan

I have tested the new section and docs change by running Expo docs website locally.

Screenshot 2021-07-09 at 16 19 38

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-apple-authentication/src/AppleAuthentication.types.ts [hunk 1]
```diff
@@ -1,12 +1,24 @@
 import { StyleProp, ViewStyle } from 'react-native';
```
Comment (bbarthec) [inline packages/expo-apple-authentication/src/AppleAuthentication.types.ts:1]:
```suggestion
import { StyleProp, ViewStyle, ViewProps } from 'react-native';
```

File: packages/expo-apple-authentication/src/AppleAuthentication.types.ts [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 
-/**
- * @hidden
- */
+// @needsAudit @docsMissing
 export type AppleAuthenticationButtonProps = {
```
Comment (bbarthec) [inline packages/expo-apple-authentication/src/AppleAuthentication.types.ts:4]: As the docs suggest we can derive the props from the `View` component, so let's do it, shall we? 😊 
```suggestion
export interface AppleAuthenticationButtonProps extends ViewProps {
```

**Review by bbarthec — CHANGES_REQUESTED** [review]
Regarding your questions:
- `FunctionComponent` === `FC` (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f09b036e3d1842821eb0c2dc2746981c515ed87a/types/react/index.d.ts#L541) so it's perfectly fine to use this alias
- regarding deriving the type from the `View` component, let's add some types inheritance to enable TS magical features 🧙

**Review by bbarthec — APPROVED** [review]
Once the CI is done I'm going to merge this 💪 
Great job, as always! 👏

**Comment by Simek** [comment]
@bbarthec JSON data and package JS files need to be regenerated, but extending the component currently broke the props section. I will need to tak a look at this first. 

Also should we add the changelog entry now for `apple-authentication`, after the `Button` extension by `View`?

**Comment by bbarthec** [comment]
@Simek, unfortunately I need you to rebuild the `build` directory 😞

**Comment by Simek** [comment]
Sure, no problem. I have done the rebuild locally already, but I want to fix the missing props issue and push everything within the same commit 😉

**Comment by bbarthec** [comment]
I think we can add some changelog entry for excluding `backgroundColor` and `borderRadius` from available `style` and for the fact that the `AppleButtonProps` now extends `ViewProps`.

**Comment by Simek** [comment]
Unfortunately, fixing the props problem was not that trivial and it will need a bit more of work. 

I had to change the way in which `AppleAuthenticationButtonProps` and it's extension has been defined, otherwise TypeDoc exports all of the inherited props with the custom ones and there is no easy way to filter out them:
Screenshot 2021-07-13 at 11 56 40

One idea to address that might be generating from code or just hardcoding the `ViewProps` list, but I think that the declaration refactor is a bit tidier way. Let me know if you will be happy with definition like this:
```ts
export declare type AppleAuthenticationButtonProps = ViewProps & {
  [...]
};
```

It works almost fine without any other code alteration, but I will need to address two more, small issue which has been introduce by types changes:
* better support for types with `Omit`, currently type is rendered just like in TS, but this should be altered a bit to be more descriptive and straight-forward
* there is an issue with Inheritance part in props section, `ViewProps` are listed, but not visible on the page for some reason, need to investigate this further.

File: packages/expo-apple-authentication/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,9 @@
 
 ### 💡 Others
 
+- Extend the `AppleAuthenticationButton` component by `View`
+- Exclude `backgroundColor` and `borderRadius` from the `style` type
```
Comment (bbarthec) [inline packages/expo-apple-authentication/CHANGELOG.md:14]: - Exclude `backgroundColor` and `borderRadius` properties from the `AppleAuthenticationButton`'s `style` prop. These two are invalid for `AppleAuthenticationButton`, but TypeScript allowed the usage of them; instead use `buttonStyle` and `cornerRadius` props repsectively. ([#13567](https://github.com/expo/expo/pull/13567) by [@Simek](https://github.com/Simek))

File: packages/expo-apple-authentication/CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,9 @@
 
 ### 💡 Others
 
+- Extend the `AppleAuthenticationButton` component by `View`
```
Comment (bbarthec) [inline packages/expo-apple-authentication/CHANGELOG.md:13]: - Extend the `AppleAuthenticationButton` component type by the `View` component type. ([#13567](https://github.com/expo/expo/pull/13567) by [@Simek](https://github.com/Simek))

**Comment by Simek** [comment]
@bbarthec Thank you for updating the changelog entries, your suggestions were more detailed and way better overall! 🙂

File: packages/expo-apple-authentication/src/AppleAuthentication.types.ts [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
+   * The border radius to use when rendering the button. This works similarly to
+   * `style.borderRadius` in other Views.
+   */
   cornerRadius?: number;
-  style?: StyleProp<ViewStyle>;
+  /**
+   * The custom style to apply to the button. Should not include `backgroundColor` or `borderRadius`
+   * properties.
+   */
+  style?: StyleProp<Omit<ViewStyle, 'backgroundColor' | 'borderRadius'>>;
```
Comment (brentvatne) [inline packages/expo-apple-authentication/src/AppleAuthentication.types.ts:27]: :+1:

**Comment by Simek** [comment]
@bbarthec Is there something else which you want me to address in this PR? It would be nice to have this merged, to prevent the conflicts with upcoming API migration PR for `expo-font` package. 🙂

**Pull request merged**